api/v1beta2/network_types.go

```diff
@@ -308,6 +315,11 @@ type VPCSpec struct {
 	// Mutually exclusive with IPAMPool.
 	CidrBlock string `json:"cidrBlock,omitempty"`
 
+	// SecondaryCidrBlocks are additional CIDR blocks to be associated when the provider creates a managed VPC.
+	// Defaults to none. Mutually exclusive with IPAMPool. This makes sense to use if, for example, you want to use
+	// a separate IP range for pods (e.g. Cilium ENI mode).
+	SecondaryCidrBlocks []VpcCidrBlock `json:"secondaryCidrBlocks,omitempty"`
```

Comment:
Should this have `+optional` since it has the `omitempty` JSON tag?

Reply:
The [docs](https://book.kubebuilder.io/reference/markers.html#difference-between--optional-and--kubebuildervalidationoptional) say

> The most reliable way in 1.x to get +optional is omitempty.

And with our current kubebuilder version, the output is no different if I add `+optional`. Since I don't want to make assumptions for the future, sure I'll add this marker here. Note that many other fields don't consistently have it 🤷.

Reply:
Thanks for the link to the docs!

Sounds like we're covered by `omitempty` and the marker is pretty much documentation for humans.